app/models/manageiq/providers/nuage/network_manager/event_parser.rb

```diff
@@ -1,10 +1,14 @@
 module ManageIQ::Providers::Nuage::NetworkManager::EventParser
   def self.event_to_hash(event, ems_id)
     event_type = "#{event['entityType']}_#{event['type'].downcase}"
+    # Nuage pushes several events with the same eventReceivedTime causing
+    # EmsEvent to ignore some of them because of the same primary key. Here we
+    # just add several millisecconds to the event.
+    time = event["eventReceivedTime"] + Time.now.usec / 1000
     {
       :source     => "NUAGE",
       :event_type => event_type,
-      :timestamp  => DateTime.strptime((event["eventReceivedTime"] / 1000).to_s, '%s').to_s,
+      :timestamp  => DateTime.strptime(time.to_s, '%Q').strftime('%F %T.%6N'),
```

Comment:
Please add [test case](https://github.com/ManageIQ/manageiq-providers-nuage/blob/master/spec/models/manageiq/providers/nuage/network_manager/event_target_parser_spec.rb) to test the final value that appears in `:timestamp` field.

Reply:
Problem is this: https://github.com/manageiq/manageiq/blob/master/app/models/ems_event.rb#L243-L249. I incorrectly assumed this is because of the DB key, but here the filter will ignore the secon occurrence of the same event with the same timestamp.

Reply:
@gberginc - Can you share an example of two different events that are being treated as duplicates?